CMakeLists.txt

```diff
@@ -47,11 +47,11 @@ endif()
 add_subdirectory(external)
 
 find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Widgets)
-find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Core Widgets Gui OpenGL)
+find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Widgets OpenGL)
```

Comment:
Why the modules have been removed?

Reply:
Are Core and Gui modules linked automatically?

Reply:
It seems to have been included and is able to compile and run properly.

Reply:
Actually they are included everywhere 
<img width="329" alt="image" src="https://user-images.githubusercontent.com/322952/209150188-a2b8d1cc-2ca9-4178-b479-218c4f887608.png">

Reply:
<img width="280" alt="image" src="https://user-images.githubusercontent.com/322952/209150303-0f535d49-b884-463e-be39-b7fc1dde6373.png">

Reply:
https://doc.qt.io/qt-6/cmake-variable-reference.html

According to this document I printed the message of `Qt6Widgets_INCLUDE_DIRS`. It may be the reason why the `gui` and `core` are deleted but the header file contains no errors.

I can change that part back if you wish.

`message(${Qt6Widgets_INCLUDE_DIRS})`

```bash
D:/Software/Qt/6.2.1/msvc2019_64/include/QtWidgetsD:/Software/Qt/6.2.1/msvc2019_64/includeD:/Software/Qt/6.2.1/msvc2019_64/include/QtCoreD:/Software/Qt/6.2.1/msvc2019_64/include/QtGui
```

Reply:
I am pretty much sure that here works a CMake mechanics. It defines targets with public properties. QtWidgets target must depend on QtCore, therefore it "inherits" all the include properties, i.e. all the include search paths ftrom QtCore

Reply:
The same must be happening with linking. 

I dunno what is better: to list all the dependencies explicitly or to use the cmake specifics

Reply:
So let's keep it the same